### PR TITLE
fix set lpmode failure issue

### PR DESCRIPTION
**Pull request by keboliu**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 1]
```diff
@@ -45,6 +66,28 @@ def set_sfp_admin_status(handle, meta, sfp_module, sfp_log_port_list, admin_stat
...
+
+    # Set low power mode status
+    lpm_mask = 1 << 8
+    if lpm_enable:
+        pmmp.eeprom_override = pmmp.eeprom_override | lpm_mask
+    else:
+        pmmp.eeprom_override = pmmp.eeprom_override & (~lpm_mask)
+
+    meta.access_cmd = SXD_ACCESS_CMD_SET
+    rc = sxd_access_reg_pmmp(pmmp, meta, 1, None, None)
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: @keboliu A named definition would be better.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 2]
```diff
@@ -19,14 +36,18 @@ def get_log_ports(handle, sfp_module):
...
 
     return log_port_list
 
-def set_sfp_admin_status(handle, meta, sfp_module, sfp_log_port_list, admin_status):
+def set_sfp_admin_status(sfp_module, admin_status):
     # Get PMAOS
     pmaos = ku_pmaos_reg()
     pmaos.module = sfp_module
+    meta = sxd_reg_meta_t()
+    meta.dev_id = 1
```
[the remark this answers is not in the document]
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: revised.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 3]
```diff
@@ -45,6 +66,28 @@ def set_sfp_admin_status(handle, meta, sfp_module, sfp_log_port_list, admin_stat
     rc = sxd_access_reg_pmaos(pmaos, meta, 1, None, None)
     assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_pmaos failed, rc = %d" % rc
 
+def set_sfp_lpmode(sfp_module, lpm_enable):
+    # Get PMMP
+    pmmp = ku_pmmp_reg()
+    pmmp.module = sfp_module
+    meta = sxd_reg_meta_t()
+    meta.dev_id = 1
+    meta.swid = 0
```
[the remark this answers is not in the document]
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: revised

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
+def get_port_admin_status_by_log_port(log_port):
+    oper_state_p = new_sx_port_oper_state_t_p()
+    admin_state_p = new_sx_port_admin_state_t_p()
+    module_state_p = new_sx_port_module_state_t_p()
+    rc = sx_api_port_state_get(handle, log_port, oper_state_p, admin_state_p, module_state_p)
+    assert rc == SXD_STATUS_SUCCESS, "sx_api_port_state_get failed, rc = %d" % rc
+
+    admin_state = sx_port_admin_state_t_p_value(admin_state_p)
+    if admin_state == SX_PORT_ADMIN_STATUS_UP:
+        return 1
```
[the remark this answers is not in the document]
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: fixed

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 1 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: fixed

**Pull request closed without merging**